modules/tensor_mechanics/src/materials/ADComputeFiniteStrainElasticStress.C

```diff
@@ -48,8 +59,12 @@ ADComputeFiniteStrainElasticStress::computeQpStress()
   // Calculate the stress in the intermediate configuration
   ADRankTwoTensor intermediate_stress;
 
-  intermediate_stress =
-      _elasticity_tensor[_qp] * (_strain_increment[_qp] + _elastic_strain_old[_qp]);
+  if (_use_old_elasticity_tensor)
+    intermediate_stress =
+        _elasticity_tensor_old[_qp] * (_strain_increment[_qp] + _elastic_strain_old[_qp]);
```

Comment:
too bad we cannot capture the old state in `_elasticity_tensor` because the current and old states have different types :-/

Reply:
I know...it is such an easy fix for regular material properties.